db/model/Link.ts

```diff
@@ -1,64 +1,57 @@
-import {
-    Entity,
-    PrimaryGeneratedColumn,
-    Column,
-    BaseEntity,
-    ManyToOne,
-    type Relation,
-    In,
-} from "typeorm"
 import { getLinkType, getUrlTarget } from "@ourworldindata/components"
-import { OwidGdocLinkJSON, Url } from "@ourworldindata/utils"
+import { Url } from "@ourworldindata/utils"
 import { GdocBase } from "./Gdoc/GdocBase.js"
 import { formatUrls } from "../../site/formatting.js"
-import { OwidGdocLinkType } from "@ourworldindata/types"
-
-@Entity("posts_gdocs_links")
-export class Link extends BaseEntity implements OwidGdocLinkJSON {
-    @PrimaryGeneratedColumn() id!: number
-    @ManyToOne(() => GdocBase, (gdoc) => gdoc.id) source!: Relation<GdocBase>
-    @Column() linkType!: OwidGdocLinkType
-    @Column() target!: string
-    @Column() queryString!: string
-    @Column() hash!: string
-    @Column() componentType!: string
-    @Column() text!: string
+import {
+    DbInsertPostGdocLink,
+    DbPlainPostGdocLink,
+    OwidGdocLinkType,
+} from "@ourworldindata/types"
+import { KnexReadonlyTransaction, knexRaw } from "../db.js"
 
-    static async getPublishedLinksTo(
-        ids: string[],
-        linkType?: Link["linkType"]
-    ): Promise<Link[]> {
-        return Link.find({
-            where: { target: In(ids), linkType },
-            relations: ["source"],
-        }).then((links) => links.filter((link) => link.source.published))
-    }
+export async function getPublishedLinksTo(
+    knex: KnexReadonlyTransaction,
+    ids: string[],
+    linkType?: OwidGdocLinkType
+): Promise<(DbPlainPostGdocLink & { sourceSlug: string })[]> {
+    const rows = await knexRaw<DbPlainPostGdocLink & { sourceSlug: string }>(
+        knex,
+        `-- sql
+        select posts_gdocs_links.*, posts_gdocs.slug as sourceSlug
+        from posts_gdocs_links
```

Comment:
Probably not worth it to try given the lack of code formatter for SQL in TS, but should we try to maintain a consistent style for our SQL? Uppercase keywords, indentation?

Reply:
There actually is one that works for selections - see the message in our slack in the #developers channel